### PR TITLE
merge ranges

### PR DESCRIPTION
**Pull request by jianwan2016**

### CODE REVIEW

File: test/HaskellWorks/Data/Network/RangeSpec.hs [hunk 1]
```diff
@@ -6,14 +6,21 @@ import HaskellWorks.Hspec.Hedgehog
...
 spec :: Spec
 spec = describe "HaskellWorks.Data.Network.RangeSpec" $ do
   describe "Range" $ do
     it "should be tested" $ requireTest $ do
-      True === True
+      R.mergeRanges [] === ([] :: [R.Range Int])
+      R.mergeRanges [R.Range 0 5] === [R.Range 0 5]
+      R.mergeRanges [R.Range 3 5, R.Range 6 7] === [R.Range 3 7]
+      R.mergeRanges [R.Range 0 5, R.Range 7 7] === [R.Range 0 5, R.Range 7 7]
+      R.mergeRanges [R.Range 'a' 'b', R.Range 'c' 'f'] === [R.Range 'a' 'f']
```
Comment (dsturnbull) [inline test/HaskellWorks/Data/Network/RangeSpec.hs]:
```suggestion
      R.mergeRanges [R.Range 'a' 'b', R.Range 'c' 'f'] === [R.Range 'a' 'f']
      R.mergeRanges [R.Range 'a' 'a', R.Range 'c' 'f'] === [R.Range 'a' 'f']
      R.mergeRanges [R.Range 'a' 'b', R.Range 'b' 'b'] === [R.Range 'a' 'b']
```

Comment (dsturnbull) [reply test/HaskellWorks/Data/Network/RangeSpec.hs]: `R.mergeRanges [R.Range [] [1], R.Range [1] [2]] === [R.Range [] [2]]`

Comment (dsturnbull) [reply test/HaskellWorks/Data/Network/RangeSpec.hs]: `R.mergeRanges [R.Range (R.Range 0 1) (R.Range 1 1), R.Range (R.Range 2 3) (R.Range 4 6)] === [R.Range (R.Range 0 6)]`

Comment (jianwan2016) [reply test/HaskellWorks/Data/Network/RangeSpec.hs]: No instance for (Enum (R.Range a0))

**Pull request merged**